### PR TITLE
fix(core - logical_layers): 11657 - wait for map readiness in more reliable manner

### PR DESCRIPTION
**Pull request by IlyaIzr**
Explanation of why i'm using such approach: https://github.com/konturio/disaster-ninja-fe/issues/101

Problem we had:
image
Now works as expected

### CODE REVIEW

**Review by andrei-kucherov — APPROVED** [review]

File: src/utils/map/waitMapEvent.ts [hunk 1]
```diff
@@ -5,10 +5,19 @@ export const waitMapEvent = <T extends keyof MapEventType>(
...
-  new Promise<MapEventType[T] & EventData>((res, rej) => {
+  new Promise<(MapEventType[T] & EventData) | void>((res, rej) => {
     try {
       map.on(event, res);
     } catch (error) {
       rej(error);
     }
   });
+
+export async function mapLoaded(map: ApplicationMap) {
```
Comment (propakov) [inline src/utils/map/waitMapEvent.ts]: do we need async word if we don't use await inside?

**Review by propakov — APPROVED** [review]
good approach! 👍

**Comment by Akiyamka** [comment]
Since we already fix layers mounting problem few times it looks to me that Generic render nice candidate for covering by tests

**Comment by IlyaIzr** [comment]
> Generic render nice candidate for covering by tests

I completly agree considering this class also asks for refactoring

**Comment by IlyaIzr** [comment]
follow up tasks were created - cover generic renderer with tests - 11725 and investigate difference between map._loaded and map.loaded() - 11726

**Pull request merged**